clippy_lints/src/manual_async_fn.rs

```diff
@@ -69,7 +69,20 @@ impl<'tcx> LateLintPass<'tcx> for ManualAsyncFn {
                     |diag| {
                         if_chain! {
                             if let Some(header_snip) = snippet_opt(cx, header_span);
-                            if let Some(ret_pos) = header_snip.rfind("->");
+                            if let Some(ret_pos) = header_snip.rfind("->").map(|rpos| {
+                                let mut rpos = rpos;
+                                let chars: Vec<char> = header_snip.chars().collect();
```

Comment:
could you refactor this into something for `utils` so https://github.com/rust-lang/rust-clippy/pull/6247/ can also use it

Reply:
Thanks for your review! I refactored this.